src/System.ComponentModel.TypeConverter/tests/Design/DesignerOptionServiceTests.cs

```diff
@@ -0,0 +1,487 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Xunit;
+
+namespace System.ComponentModel.Design.Tests
+{
+    public class DesignerOptionServiceTests
+    {
+        [Fact]
+        public void CreateOptionCollection_CreateMultipleTimes_ReturnsExpected()
+        {
+            var service = new TestDesignerOptionService();
+
+            DesignerOptionService.DesignerOptionCollection options1 = service.DoCreateOptionCollection(service.Options, "name", "value");
+            Assert.Equal(0, options1.Count);
+            Assert.NotEmpty(options1.Properties);
+            Assert.Same(options1.Properties, options1.Properties);
+            Assert.Same(service.Options, options1.Parent);
+
+            DesignerOptionService.DesignerOptionCollection options2 = service.DoCreateOptionCollection(service.Options, "name", "value");
+            Assert.Equal(2, service.Options.Count);
+            Assert.Equal(0, options2.Count);
+            Assert.NotEmpty(options2.Properties);
+            Assert.Same(options2.Properties, options2.Properties);
+            Assert.Same(service.Options, options2.Parent);
+
+            Assert.Equal(new DesignerOptionService.DesignerOptionCollection[] { options1, options2 }, service.Options.Cast<object>());
+        }
+
+        [Fact]
+        public void CreateOptionCollection_NullParent_ThrowsArgumentNullException()
+        {
+            var service = new TestDesignerOptionService();
+            AssertExtensions.Throws<ArgumentNullException>("parent", () => service.DoCreateOptionCollection(null, "name", "value"));
+        }
+
+        [Fact]
+        public void CreateOptionCollection_NullName_ThrowsArgumentNullException()
+        {
+            var service = new TestDesignerOptionService();
+            AssertExtensions.Throws<ArgumentNullException>("name", () => service.DoCreateOptionCollection(service.Options, null, "value"));
+        }
+
+        [Fact]
+        public void CreateOptionCollection_EmptyName_ThrowsArgumentException()
+        {
+            var service = new TestDesignerOptionService();
+            AssertExtensions.Throws<ArgumentException>("name.Length", () => service.DoCreateOptionCollection(service.Options, string.Empty, "value"));
+        }
+
+        [Fact]
+        public void Options_Get_ReturnsExpected()
+        {
+            var service = new TestDesignerOptionService();
+            DesignerOptionService.DesignerOptionCollection options = service.Options;
+            Assert.Same(options, service.Options);
+
+            Assert.Equal(0, options.Count);
+            Assert.Empty(options.Properties);
+            Assert.Same(options.Properties, options.Properties);
```

Comment:
Same. There is actually a subtle bug here, as the cached properties stays the same even though the children may change.

See `Properties_GetBeforeAddingChild_ReturnsEmpty` and `Properties_GetAfterAddingChild_ReturnsNonEmpty`